pkg/backporter/backporter.go

```diff
@@ -0,0 +1,306 @@
+						{{range $index, $pr := $clone.PRs }}
+							{{ if $index}},{{end}}
+							<a href = {{ $pr.Type.URL }}/{{$pr.Org}}/{{$pr.Repo}}/pull/{{$pr.Num}} target="_blank"> {{$pr.Org}}/{{$pr.Repo}}#{{$pr.Num}}</a>
+						{{end}}
+					</td>
+				</tr>
+			{{ end }}
+		{{ else }}
+			<tr> <td colspan=4 style="text-align:center;"> No clones found! </td></tr>
+		{{ end }}
+		</tbody>
+	</table>`
+
+// HandlerFuncWithErrorReturn allows returning errors to be logged
+type HandlerFuncWithErrorReturn func(http.ResponseWriter, *http.Request) error
+
+type wrapper struct {
+	Bug    *bugzilla.Bug
+	Clones []*bugzilla.Bug
+	Parent *bugzilla.Bug
+	PRs    []bugzilla.ExternalBug
+}
+
+func getTemplate(tConstructor string) (*template.Template, error) {
+	tmpl, err := template.New("").Parse(tConstructor)
+	if err != nil {
+		return nil, fmt.Errorf("Incorrect template for page %s", tConstructor)
+	}
+	return tmpl, nil
+}
+
+//Writes an HTML page, prepends header in htmlPageStart and appends header from htmlPageEnd around tConstructor.
+func writePage(w http.ResponseWriter, title string, tConstructor string, data interface{}) error {
+	fmt.Fprintf(w, htmlPageStart, title)
+	body, err := template.New(title).Parse(tConstructor)
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "Incorrect template %s: %v", tConstructor, err)
+		return err
+	}
+	if err := body.Execute(w, data); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		fmt.Fprintf(w, "%s: %v", http.StatusText(http.StatusInternalServerError), err)
+		return err
+	}
+	fmt.Fprintln(w, htmlPageEnd)
+	return nil
+}
+
+// GetLandingHandler will return a simple bug search page
+func GetLandingHandler() HandlerFuncWithErrorReturn {
+	return func(w http.ResponseWriter, req *http.Request) error {
+		writePage(w, "Home", emptyTemplateConstructor, nil)
+		return nil
+	}
+}
+
+// GetBugHandler returns a function which populates the response with the details of the bug
+// Returns bug details in JSON format
+func GetBugHandler(client bugzilla.Client) HandlerFuncWithErrorReturn {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		if r.Method == "GET" {
+			bugIDStr := r.URL.Query().Get(BugIDQuery)
+			bugID, err := strconv.Atoi(bugIDStr)
+			if bugIDStr == "" || err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				fmt.Fprintf(w, "%s query missing or incorrect", BugIDQuery)
+				return err
+			}
+			bugInfo, err := client.GetBug(bugID)
+			if err != nil {
+				w.WriteHeader(http.StatusBadRequest)
+				w.Write([]byte("Bug ID not found"))
+				return err
+			}
+			jsonBugInfo, err := json.MarshalIndent(*bugInfo, "", "  ")
+			if err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
+				fmt.Fprintf(w, "failed to marshal bugInfo to JSON: %v", err)
+				// logger.WithError(err).Errorf("failed to marshal bugInfo to JSON")
+				return err
+			}
+			w.WriteHeader(http.StatusOK)
+			w.Write(jsonBugInfo)
+		} else {
```

Comment:
The idiomatic way to do thing like this in go is to return early, rather than writing arrow code https://blog.codinghorror.com/flattening-arrow-code/

So here this would mean:
```
if method != "GET" {
  error handling
}
// business logic here
```

rather than
```
if method == "GET" {
 a
 large
 amount
 of
 business
 logic
else {
  error handling
}
```

Because by the time the reader of the code reaches the `else`, they will have forgotten what the condition checks for

Reply:
Thank you for pointing that out